docs/ros-snap.md

```diff
@@ -0,0 +1,354 @@
+  scripts/listener_node
+  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+)
+
+install(FILES
+  talk_and_listen.launch
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
+```
+
+Now for `scripts/listener_node`, we have this:
+
+```python
+#!/usr/bin/env python
+
+import rospy
+from std_msgs.msg import String
+
+def callback(data):
+    rospy.loginfo('I heard %s', data.data)
+
+def listener():
+    rospy.init_node('listener')
+
+    rospy.Subscriber('babble', String, callback)
+
+    rospy.spin()
+
+if __name__ == '__main__':
+    listener()
+```
+
+Note that the listener subscribes to the `babble` topic, but the talker
+publishes on `chatter`. We need to make sure we account for that in
+`talk_and_listen.launch`:
+
+```xml
+<launch>
+    <node name = "talker" pkg = "talker" type = "talker_node"
+          output = "screen" />
+
+    <node name = "listener" pkg = "listener" type = "listener_node"
+          output = "screen">
+        <remap from = "babble" to = "chatter" />
+    </node>
+</launch>
+```
+
+The "listener" is now complete.
+
+
+### Verify functionality
+
+You can verify that everything works by getting in the workspace root and
+running:
+
+    $ catkin_make
+    $ catkin_make install
+    $ roslaunch listener talk_and_listen.launch
+
+You should see them communicating, looking something like this:
+
+    ...
+    [INFO] [WallTime: 1449512660.316140] I heard Hello world 2
+    [ INFO] [1449512660.415941529]: Hello world 3
+    [INFO] [WallTime: 1449512660.416330] I heard Hello world 3
+    [ INFO] [1449512660.515954882]: Hello world 4
+    [INFO] [WallTime: 1449512660.516307] I heard Hello world 4
+    [ INFO] [1449512660.615954473]: Hello world 5
+    [INFO] [WallTime: 1449512660.616306] I heard Hello world 5
+    ...
+
+### Put it all in a .snap
+
+Create a `snapcraft.yaml` in the workspace root, containing the following:
+
+```yaml
+name: ros-talker-and-listener
+version: 1.0
+vendor: example <example@example.com>
+summary: ROS Example
+description: Contains talker/listener ROS packages and a .launch file.
+icon: icon.svg
+
+binaries:
+  launch_project:
+    exec: roslaunch listener talk_and_listen.launch
+
+parts:
+  foo:
+    plugin: catkin
+    source: .
+    catkin-packages:
+      - talker
+      - listener
+    stage-packages:
+      - ros-indigo-ros-core
+```
+
+Most of this file should look familiar to you, but let's focus on a few specific
+pieces.
+
+```yaml
+# ...
+binaries:
+  launch_project:
+    exec: roslaunch listener talk_and_listen.launch
+# ...
+```
+
+Even though the `talker` and `listener` packages will be installed, they'll be
+within a confined ROS installation, so the user won't be able to simply call
+`rosrun` or `roslaunch`. Instead, you have control over how your .snap is used,
+and here we specify that we only want a single binary, called "launch_project",
+which results in the `roslaunch` call you see. If this seems confusing now, it
+might make more sense when we actually use it.
```

Comment:
nitpicking here, I think you should mention that we'll give it a try later in the document.
"it WILL make more sense later in this document when we actually use it."